### ISSUE TITLE
[NOT AN ISSUE] Code for training the OCR model (CNN-RNN)

### ISSUE THREAD

**Issue by johnlockejrr** [qurator-spk/eynollah#208]
Is there any code for training/finetuning the OCR model? Or is used a `TrOCR` model as I can see in the `eynollah` source (`model_eynollah_ocr_trocr_20250919`)? Even though I see a `model_eynollah_ocr_cnnrnn_20250930` model also.

**Comment by johnlockejrr** [comment]
And by the way, any reason to stick with `tensorflow`? I refactored the `sbb_pixelwise_segmentation` training sources to `pythorch` and is damn faster than the `tensorflow` version and easy to port.

**Comment by vahidrezanezhad** [comment]
@johnlockejrr The CNN-RNN OCR model is our own implementation, and its training script will be integrated into the training tool as soon as possible. For the transformer-based model, we utilized TrOCR for training.

**Comment by vahidrezanezhad** [comment]
> And by the way, any reason to stick with `tensorflow`? I refactored the `sbb_pixelwise_segmentation` training sources to `pythorch` and is damn faster than the `tensorflow` version and easy to port.

 I'm aware of your work on refactoring the training sources to PyTorch — really great job! I truly appreciate your effort, and I plan to use it as a reference when converting our models to PyTorch :)

The reason we still use TensorFlow is that it has been our main framework since the beginning — we used it for both training and tool development. We also had to focus on improving the OCR and layout modules while keeping the codebase stable and maintainable. So, migrating to PyTorch wasn’t our top priority at that stage.

That said, it’s definitely part of our plan to move everything to PyTorch to take advantage of the performance gains you mentioned.

**Comment by cneud** [comment]
>  I refactored the sbb_pixelwise_segmentation training sources to pythorch and is damn faster than the tensorflow version and easy to port.

@johnlockejrr do you happen to have changes related to this already in a fork/branch? That would be extremely helpful!

**Comment by johnlockejrr** [comment]
Yes, when I got home I will push the changelog to my github

**Comment by johnlockejrr** [comment]
> > I refactored the sbb_pixelwise_segmentation training sources to pythorch and is damn faster than the tensorflow version and easy to port.
> 
> [@johnlockejrr](https://github.com/johnlockejrr) do you happen to have changes related to this already in a fork/branch? That would be extremely helpful!

https://github.com/johnlockejrr/sbb_pytorch/blob/main/CHANGELOG.md
https://github.com/johnlockejrr/sbb_pytorch/blob/main/TECHNICAL_CHANGELOG.md

**Comment by cneud** [comment]
> https://github.com/johnlockejrr/sbb_pytorch/blob/main/CHANGELOG.md 
> https://github.com/johnlockejrr/sbb_pytorch/blob/main/TECHNICAL_CHANGELOG.md

Thanks you! Unfortunately the repo is not accessible yet, maybe it is set to private?

**Comment by johnlockejrr** [comment]
> > https://github.com/johnlockejrr/sbb_pytorch/blob/main/CHANGELOG.md
> > https://github.com/johnlockejrr/sbb_pytorch/blob/main/TECHNICAL_CHANGELOG.md
> 
> Thanks you! Unfortunately the repo is not accessible yet, maybe it is set to private?

Yes, I overlooked this, was a private test that's why I kept the visibility as such. Try now.

**Comment by johnlockejrr** [comment]
And, by the way, I just tested the success of porting it from tensorflow to pythorch (and it works), I worked a little messy, all code in the same directory etc. ;)
These days if I got some free time I will port the current "train" module from `eynollah` and play a little dirrectly with `eynollah`
